tgui/packages/tgui/interfaces/PowerMonitor/index.tsx

```diff
@@ -40,11 +96,25 @@ export const PowerMonitor = (_props, context) => {
           <Stack.Item grow={1}>
             <Section fill scrollable>
               <Table>
-                <Table.Row header>{isDataForApc(data) && <PowerMonitorApcTableHeader />}</Table.Row>
-                {isDataForApc(data) && <PowerMonitorApcTableRows search={search} />}
+                <Table.Row header>
+                  {
+                    isDataForApc(data)
+                    && <PowerMonitorApcTableHeader
+                      state={apcSortBy}
+                      setSortBy={(field) => onSetApcHeaderSort(field, apcSortBy)} />
+                  }
+                </Table.Row>
+                {isDataForApc(data) && <PowerMonitorApcTableRows sortState={apcSortBy} search={search} />}
 
-                <Table.Row header>{isDataForSmes(data) && <PowerMonitorSmesTableHeader />}</Table.Row>
-                {isDataForSmes(data) && <PowerMonitorSmesTableRows search={search} />}
+                <Table.Row header>
+                  {
+                    isDataForSmes(data)
+                    && <PowerMonitorSmesTableHeader
+                      state={smesSortBy}
+                      setSortBy={(field) => onSetSmesHeaderSort(field, smesSortBy)} />
+                  }
+                </Table.Row>
+                {isDataForSmes(data) && <PowerMonitorSmesTableRows sortState={smesSortBy} search={search} />}
```

Comment:
Honestly these are two components at this point, that happen to share an interface.

Rather than doing the conditional inside the `Table.Row` and around the `...TableRows` components individually, do it around the combo then return a fragment with the needed components.

Reply:
Have restructured it to be one component per interface to make it less messy